common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -364,18 +364,18 @@ public void record(Object hint) {
         }
 
         private void record0(Object hint, int recordsToSkip) {
-            if (creationRecord != null) {
+            if (creationRecord != null && MAX_RECORDS > 0) {
```

Comment:
@louxiu I think this is not correct as this change behaviour. Before the user could use MAX_RECORDS = 0 to basically enforce there is no limit at all.  Now we not enter the block all. Can you explain why you think this is correct here ?